scalding-commons/src/main/scala/com/twitter/scalding/commons/source/VersionedKeyValSource.scala

```diff
@@ -124,6 +124,25 @@ class VersionedKeyValSource[K, V](val path: String, val sourceVersion: Option[Lo
       }
     }
 
+  def sinkExists(mode: Mode) =
+    sinkVersion match {
+      case Some(version) =>
+        mode match {
+          case Test(buffers) =>
+            buffers(this) map { !_.isEmpty } getOrElse false
+
+          case HadoopTest(conf, buffers) =>
+            buffers(this) map { !_.isEmpty } getOrElse false
+
+          case m: HadoopMode =>
+            val conf = new JobConf(m.jobConf)
+            val store = sink.getStore(conf)
+            store.hasVersion(version)
+          case _ => sys.error(s"Unknown mode $mode")
```

Comment:
Maybe use a `Try`/`Maybe` instead of throwing an error.  How likely is it that there will be a mode other than those three?